src/qibolab/instruments/qm.py

```diff
@@ -373,22 +373,22 @@ def __post_init__(self):
         if self.raw_adc:
             self.adc_stream = declare_stream(adc_trace=True)
 
-        if self.threshold is not None:
+        if self.threshold is not None and self.angle is not None:
             self.shot = declare(bool)
             self.shots = declare_stream()
             self.cos = np.cos(self.angle)
             self.sin = np.sin(self.angle)
 
     def save(self):
         """QUA instruction to save acquired results from variables to streams."""
-        save(self.I, self.I_stream)
-        save(self.Q, self.Q_stream)
-        if self.threshold is not None:
-            save(self.shot, self.shots)
+        qua.save(self.I, self.I_stream)
+        qua.save(self.Q, self.Q_stream)
+        if self.shot is not None:
+            qua.save(self.shot, self.shots)
 
     def classify_shots(self):
         """QUA instruction to classify shots in real time and save the result to a variable."""
-        if self.threshold is not None:
+        if self.shot is not None:
```

Comment:
Are you sure? You're only assigning to it if it's not `None`, this is slightly counter-intuitive...

Reply:
It would work as it was because `self.shot` is `None` if `self.threshold is None` but you are right that it was not very intuitive so I fixed it.